src/v/datalake/tests/catalog_schema_manager_test.cc

```diff
@@ -305,6 +305,12 @@ TEST_F(CatalogSchemaManagerTest, AcceptsValidTypePromotion) {
       nested_field::create(18, "some_field", field_required::yes, int_type{}));
     create_table(original_type);
 
+    // create original schema
+    auto ensure_orig_res
```

Comment:
Please add context into the commit message about what was broken about this. At least, it isn't clear IMO why we need this step here vs relying on create_table (and e.g. why other tests don't need a similar change)

Reply:
Reading it again ... it was correct before. Dropped the commit. I didn't see the create table and assumed the test gets an implicit table with `test_nested_schema_type` schema and the one we modify in test wasn't actually applied. 🤦‍♂️